src/code/z_parameter.c

```diff
@@ -2267,115 +2273,134 @@ void Inventory_ChangeAmmo(s16 item, s16 ammoChange) {
+                    return true;
                 } else {
-                    return 0;
+                    return false;
                 }
             } else {
-                if (gSaveContext.unk_13F0 == 7) {
-                    return 1;
+                if (gSaveContext.magicState == MAGIC_STATE_CONSUME_LENS) {
+                    return true;
                 } else {
-                    return 0;
+                    return false;
                 }
             }
-        case 4:
-            if ((gSaveContext.unk_13F0 == 0) || (gSaveContext.unk_13F0 == 7)) {
-                if (gSaveContext.unk_13F0 == 7) {
+
+        case MAGIC_CONSUME_WAIT_PREVIEW:
+            // Sets consume target but waits to consume.
+            // Preview consumption with a yellow bar
+            if ((gSaveContext.magicState == MAGIC_STATE_IDLE) ||
+                (gSaveContext.magicState == MAGIC_STATE_CONSUME_LENS)) {
+                if (gSaveContext.magicState == MAGIC_STATE_CONSUME_LENS) {
                     globalCtx->actorCtx.lensActive = false;
                 }
-                gSaveContext.unk_13F8 = gSaveContext.magic - arg1;
-                gSaveContext.unk_13F0 = 4;
-                return 1;
+                gSaveContext.magicTarget = gSaveContext.magic - amount;
+                gSaveContext.magicState = MAGIC_STATE_METER_FLASH_2;
+                return true;
             } else {
                 Audio_PlaySoundGeneral(NA_SE_SY_ERROR, &gSfxDefaultPos, 4, &gSfxDefaultFreqAndVolScale,
                                        &gSfxDefaultFreqAndVolScale, &gSfxDefaultReverb);
-                return 0;
+                return false;
             }
-        case 5:
-            if (gSaveContext.unk_13F4 >= gSaveContext.magic) {
-                gSaveContext.unk_13F8 = gSaveContext.magic + arg1;
 
-                if (gSaveContext.unk_13F8 >= gSaveContext.unk_13F4) {
-                    gSaveContext.unk_13F8 = gSaveContext.unk_13F4;
+        case MAGIC_ADD:
+            // Sets target for magic to increase to
+            if (gSaveContext.magic <= gSaveContext.magicCapacityDrawn) {
+                gSaveContext.magicTarget = gSaveContext.magic + amount;
+
+                if (gSaveContext.magicTarget >= gSaveContext.magicCapacityDrawn) {
+                    gSaveContext.magicTarget = gSaveContext.magicCapacityDrawn;
                 }
 
-                gSaveContext.unk_13F0 = 10;
-                return 1;
+                gSaveContext.magicState = MAGIC_STATE_ADD;
+                return true;
             }
             break;
     }
 
-    return 0;
+    return false;
 }
 
-void Interface_UpdateMagicBar(GlobalContext* globalCtx) {
+void Magic_Update(GlobalContext* globalCtx) {
     static s16 sMagicBorderColors[][3] = {
         { 255, 255, 255 },
         { 150, 150, 150 },
-        { 255, 255, 150 },
-        { 255, 255, 50 },
+        { 255, 255, 150 }, // unused color
+        { 255, 255, 50 },  // unused color
```

Comment:
```suggestion
        { 255, 255, 150 }, // unused
        { 255, 255, 50 },  // unused
```
Since we already know they're colours from the array name.